File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    area = brown + yellow
    
    for i in range(1, area):
         if area % i == 0:
                b = int(i * 2 + (area / i) * 2 - 4)
                y = int(area - b)
                if brown == b and yellow == y:
                    answer = [int(area / i),i]
                    break
                    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = {0, 0};
        int area = brown + yellow;
        
        for (int i = 1; i < area/2+1; i++) {
            if (area % i == 0) {
                int j = area / i;
                if (i*2 + j*2 - 4 == brown) {
                    answer[0] = j;
                    answer[1] = i;
                    break;
                }
            }
        }
            
        return answer;
    }
}